internal/contour/snapshot.go

```diff
@@ -40,7 +52,70 @@ func NewSnapshotHandler(c cache.SnapshotCache, resources []ResourceCache, logger
 	}
 }
 
-// OnChange is called when the DAG is rebuilt
-// and a new snapshot is needed.
+// OnChange is called when the DAG is rebuilt and a new snapshot is needed.
 func (s *SnapshotHandler) OnChange(root *dag.DAG) {
+
```

Comment:
You can also make the ordering robust by forwarding the DAG change to the resources here (also remove them from the builder in serve.go). This guarantees that all the resources are updated before the snapshot is made.

```Go
for _, r := range s.resources {
    r.OnChange(root)
}
```